README.md

```diff
@@ -11,24 +18,111 @@
           ``'"`
 ```
 
-OpenSquirrel chooses a _modular_, over a _configurable_, approach to prepare and optimize quantum circuits for heterogeneous target architectures.
+OpenSquirrel is a quantum compiler that chooses a _modular_, over a _configurable_, approach to prepare and optimize quantum circuits for heterogeneous target architectures.
 
 It has a user-friendly interface and is straightforwardly extensible with custom-made readers, compiler passes, and exporters.
 As a quantum circuit compiler, it is fully aware of the semantics of each gate and arbitrary quantum gates can be constructed manually.
 It understands the quantum programming language cQASM 3 and will support additional quantum programming languages in the future.
 It is developed in modern Python and follows best practices.
 
-
 ## Installation
 
+We encourage installing OpenSquirrel via `pip`:
+
 ```shell
 $ pip install opensquirrel
 ```
+To install the dependencies to run the examples on `jupyter`, install:
+
+```shell
+$ pip install opensquirrel[examples]
+```
+The tutorials can be found [here](https://github.com/QuTech-Delft/OpenSquirrel/tree/develop/example).
+
+## Getting started
+
+Essentially, compiling a circuit in OpenSquirrel requires,
+1. Defining and building a quantum circuit using the openSquirrel IR directly or using a `CQASM` string
+2. Modifying and traversing the circuit
+3. Writing the circuit to assembly
+
+The circuit can be built with the `CircuitBuilder` class. Here is an example of building a single qubit circuit that
+includes a Hadamard gate, a Z gate, a Y gate, and a rotation X gate.
+
+```python
+import math
+from opensquirrel.circuit_builder import CircuitBuilder
+from opensquirrel.ir import Qubit, Float
+
+# Tell the circuit builder how you want your circuit
+builder = CircuitBuilder(qubit_register_size=1)
+builder.H(Qubit(0))
+builder.Z(Qubit(0))
+builder.Y(Qubit(0))
+builder.Rx(Qubit(0), Float(math.pi / 3))
+
+# Get the circuit from the circuit builder
+circuit = builder.to_circuit()
+```
+This can also be done directly using a `CQASM3` string as shown below,
+```python
+from opensquirrel.circuit import Circuit
```

Comment:
```suggestion
circuit = builder.to_circuit()
```

This can also be done directly using a `CQASM3` string as shown below:

```python
from opensquirrel.circuit import Circuit
```